### PR TITLE
Add generalized community name and description to onboarding slides

### PR DESCRIPTION
**Pull request by benhalpern**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Replacing some hardcoded values in onboarding.

Pretty small change, but touches 7 files due to snapshot tests and that it adds a new prop to the components.

### CODE REVIEW

File: app/javascript/onboarding/__tests__/__snapshots__/Onboarding.test.jsx.snap [hunk 1]
```diff
@@ -348,10 +348,8 @@ preact-render-spy (1 nodes)
             alt="DEV"
            />
         </figure>
-        <h1 class="introduction-title">firstname lastname— welcome to DEV!</h1>
-        <h2 class="introduction-subtitle">
-          DEV is where programmers share ideas and help each other grow.
-        </h2>
+        <h1 class="introduction-title">firstname lastname— welcome to Test!</h1>
+        <h2 class="introduction-subtitle">Wouldn't you like to knoww..</h2>
```
Comment (vaidehijoshi) [inline app/javascript/onboarding/__tests__/__snapshots__/Onboarding.test.jsx.snap:352]: Hm, how come the styles for “Wouldn’t you like to knoww..” change? 

It doesn’t look like you fiddled with the class so I wouldn’t expect a style change...

File: app/javascript/onboarding/Onboarding.jsx [hunk 2]
```diff
@@ -69,3 +71,7 @@ export default class Onboarding extends Component {
     return <main className="onboarding-body">{this.slides[currentSlide]}</main>;
   }
 }
+
+Onboarding.propTypes = {
+  communityConfig: PropTypes.shape.isRequired,
```
Comment (vaidehijoshi) [inline app/javascript/onboarding/Onboarding.jsx:76]:
```suggestion
  communityConfig: PropTypes.shape({
    communityName: PropTypes.string.isRequired,
    communityDescription: PropTypes.string.isRequired
  })
```
TIL: about `PropTypes.shape`! Looking at [the docs](https://reactjs.org/docs/typechecking-with-proptypes.html#proptypes), I think we should be specifying the keys in the object, right?

**Comment by benhalpern** [comment]
Thanks for the shapes suggestion @vaidehijoshi tbh I basically just put whatever the linter would stop yelling at me for. So thanks for the improved suggestion.

**Comment by vaidehijoshi** [comment]
@benhalpern LOL, thanks for inadvertently introducing me to it! :)

File: app/javascript/onboarding/components/IntroSlide.jsx [hunk 3]
```diff
@@ -187,6 +195,7 @@ IntroSlide.propTypes = {
   next: PropTypes.func.isRequired,
   slidesCount: PropTypes.number.isRequired,
   currentSlideIndex: PropTypes.func.isRequired,
+  communityConfig: PropTypes.shape.isRequired,
```
Comment (benhalpern) [inline app/javascript/onboarding/components/IntroSlide.jsx:198]:
```suggestion
  communityConfig: PropTypes.shape({
    communityName: PropTypes.string.isRequired,
    communityDescription: PropTypes.string.isRequired
  }),
```

File: app/javascript/onboarding/components/ProfileForm.jsx [hunk 4]
```diff
@@ -156,6 +164,7 @@ ProfileForm.propTypes = {
   next: PropTypes.func.isRequired,
   slidesCount: PropTypes.number.isRequired,
   currentSlideIndex: PropTypes.func.isRequired,
+  communityConfig: PropTypes.shape.isRequired,
```
Comment (benhalpern) [inline app/javascript/onboarding/components/ProfileForm.jsx:167]:
```suggestion
  communityConfig: PropTypes.shape({
    communityName: PropTypes.string.isRequired,
    communityDescription: PropTypes.string.isRequired
  }),
```

File: app/javascript/onboarding/Onboarding.jsx [hunk 5]
```diff
@@ -69,3 +71,10 @@ export default class Onboarding extends Component {
     return <main className="onboarding-body">{this.slides[currentSlide]}</main>;
   }
 }
+
+Onboarding.propTypes = {
+  communityConfig: PropTypes.shape({
+    communityName: PropTypes.string.isRequired,
+    communityDescription: PropTypes.string.isRequired
+  })
```
Comment (rhymes) [inline app/javascript/onboarding/Onboarding.jsx:79]: you can add `isRequired` to the shape as well `shape().isRequired`

**Pull request merged**